providers/fips/fipsprov.c

```diff
@@ -196,12 +209,18 @@ static int fips_get_params(void *provctx, OSSL_PARAM params[])
     p = OSSL_PARAM_locate(params, OSSL_PROV_PARAM_STATUS);
     if (p != NULL && !OSSL_PARAM_set_int(p, ossl_prov_is_running()))
         return 0;
-    p = OSSL_PARAM_locate(params, OSSL_PROV_PARAM_SECURITY_CHECKS);
-    if (p != NULL && !OSSL_PARAM_set_int(p, fgbl->fips_security_checks))
-        return 0;
-    p = OSSL_PARAM_locate(params, OSSL_PROV_PARAM_TLS1_PRF_EMS_CHECK);
-    if (p != NULL && !OSSL_PARAM_set_int(p, fgbl->fips_tls1_prf_ems_check))
-        return 0;
+
+#define FIPS_FEATURE_GET(fgbl, pname, field)                            \
+    p = OSSL_PARAM_locate(params, pname);                               \
+    if (p != NULL && !OSSL_PARAM_set_int(p, fgbl->field.enabled))       \
```

Comment:
I guess this set_int will work?

Reply:
Should do, the field will promote to int without loss.
The call will do any other required conversions.